geo-types/src/coordinate.rs

```diff
@@ -309,7 +309,7 @@ where
 
     const DIMENSIONS: usize = 2;
 
-    fn generate(generator: impl Fn(usize) -> Self::Scalar) -> Self {
+    fn generate(mut generator: impl FnMut(usize) -> Self::Scalar) -> Self {
```

Comment:
calling all semver lawyers: is this considered a breaking change for geo-types?

Reply:
I don't think so. It's more general and if there was a way for the argument type to leak into the outside code (*), type inference changes are generally considered benign.

(*) E.g. `x` here gets inferred as `()` by `rustc`:

```rust
fn f(g: impl Fn()) {}

fn main() {
    let x;
    f(x);
    x = todo!();
}
```

Reply:
Actually, no, it's a breaking change because `rstar::Point` is already in our public API. So if user code depends on `rstar 0.8`, they'll have to upgrade to `0.9`.

Reply:
Is this an argument for moving the `geo-types` trait impls to the rstar crate?

Reply:
Just that it's a real pain to bump geo-types, since so many depend on it.

Reply:
> Is this an argument for moving the geo-types trait impls to the rstar crate?

That's a pretty sensible idea.